components/infobox/wikis/starcraft/infobox_team_custom.lua

```diff
@@ -0,0 +1,336 @@
+
+local Injector = require('Module:Infobox/Widget/Injector')
+local Cell = require('Module:Infobox/Widget/Cell')
+local Title = require('Module:Infobox/Widget/Title')
+local Breakdown = require('Module:Infobox/Widget/Breakdown')
+
+local Condition = require('Module:Condition')
+
+local ConditionTree = Condition.Tree
+local ConditionNode = Condition.Node
+local Comparator = Condition.Comparator
+local BooleanOperator = Condition.BooleanOperator
+local ColumnName = Condition.ColumnName
+
+local doStore = true
+
+local CustomTeam = Class.new()
+
+local CustomInjector = Class.new(Injector)
+local _LANGUAGE = mw.language.new('en')
+
+local _earnings = 0
+local _EARNINGS_MODES = {team = 'team'}
+local _ALLOWED_PLACES = {'1', '2', '3', '4', '3-4'}
+local _MAXIMUM_NUMBER_OF_PLAYERS_IN_PLACEMENTS = 35
+local _PLAYER_EARNINGS_ABBREVIATION = '<abbr title="Earnings of players while on the team">Player earnings</abbr>'
+
+local _args
+local _team
+
+function CustomTeam.run(frame)
+	local team = Team(frame)
+	_team = team
+	_args = team.args
+
+	team.getWikiCategories = CustomTeam.getWikiCategories
+	team.addToLpdb = CustomTeam.addToLpdb
+	team.createWidgetInjector = CustomTeam.createWidgetInjector
+
+	return team:createInfobox(frame)
+end
+
+function CustomInjector:addCustomCells(widgets)
+	table.insert(widgets, Cell{
+		name = 'Gaming Director',
+		content = {_args['gaming director']}
+	})
+
+	return widgets
+end
+
+function CustomInjector:parse(id, widgets)
+	if id == 'earnings' then
+		local earningsWhileOnTeam
+		_earnings, earningsWhileOnTeam = CustomTeam.calculateEarnings(_args)
+		local earningsDisplay
+		if _earnings == 0 then
+			earningsDisplay = nil
+		else
+			earningsDisplay = '$' .. _LANGUAGE:formatNum(_earnings)
+		end
+		local earningsFromPlayersDisplay
+		if earningsWhileOnTeam > 0 then
+			earningsFromPlayersDisplay = '$' .. _LANGUAGE:formatNum(earningsWhileOnTeam)
+		end
+		return {
+			Cell{name = 'Approx. Total Winnings', content = {earningsDisplay}},
+			Cell{name = _PLAYER_EARNINGS_ABBREVIATION, content = {earningsFromPlayersDisplay}},
+		}
+	elseif id == 'achievements' then
+		table.insert(widgets, Cell{name = 'Solo Achievements', content = {_args['solo achievements']}})
+		--need this ABOVE the history display and below the
+		--achievements display, hence moved it here
+		local playerBreakDown = CustomTeam.playerBreakDown(_args)
+		if playerBreakDown.playernumber then
+				table.insert(widgets, Title{name = 'Player Breakdown'})
+				table.insert(widgets, Cell{name = 'Number of players', content = {playerBreakDown.playernumber}})
+				table.insert(widgets, Breakdown{content = playerBreakDown.display, classes = {'infobox-center'}})
+		end
+	elseif id == 'history' then
+		local index = 1
+		while(not String.isEmpty(_args['history' .. index .. 'title'])) do
+			table.insert(widgets, Cell{
+				name = _args['history' .. index .. 'title'],
+				content = {_args['history' .. index]}
+			})
+			index = index + 1
+		end
```

Comment:
`pairsByPrefix`

Reply:
wouldn't work
it checks if `args.historyXtitle` exists
`pairsByPrefix` would only check if `args.historyX` exists

Reply:
Doesn't work to check for `historyX` instead`. Just looking at the code, it would expect them to go hand in hand.

Reply:
if `historyXtitle` is missing while `historyX` is set it would cause errors (mandatory field of cell not set)
the other way around it will just not display the cell

could do something like
```lua
for historyKey, historyItem in Table.iter.pairsByPrefix(_args, 'history') do
	local historyTitle = _args[historyKey .. 'title']
	if String.isNotEmpty(historyTitle) then
		table.insert(......)
	end
end


```

Reply:
Ok

Reply:
so should i do what i suggested in above comment or not?

Reply:
We can leave it out. Meant to click resolve, must have missclicked.